### DISCUSSION TITLE
extracting data from messy home cage tracking data

### DISCUSSION THREAD

**Discussion by lustrauch** [talmolab/sleap#2014] [Help!]
Dear Sleap Team, 

Thank you for creating this great tool! 

I have a few questions regarding the extraction of the data. I am for the first time training a multi-animal model where I have one animal with an ID (the dam) and eight pups that we did not assign any track IDs to because it is almost impossible to differentiate between them. For all of them, we used the same simple skeleton as we just wanted to see if we even have any chance of tracking the pups in this messy environment. 
The predictions look quite good now but when I extract the data I somehow get only an Excel sheet containing the frames where the dam (so the animal with the ID) was identified. 
1) My first question is quite general: is it correct for a multiple animal models to use the same skeleton (same node points) for all the animals or should they be unique eg neck1 and neck2? 
2) How do I get the tracking data from all the animals when they are not assigned IDs? 
3) Do you have any suggestions to handle quite messy home cage tracking data? We are video monitoring the animals from P0-P21 to investigate how early life stress (using the limited bedding model) influences the behaviour of the pups and dam. 

Thank you and best regards,
Luna 
![pastedImage-1](https://github.com/user-attachments/assets/89ba3f53-5f5c-4676-8366-d1806b8192a1)
![pastedImage](https://github.com/user-attachments/assets/a41bf509-dfe4-4e53-
<img width="1073" alt="Screenshot 2024-11-04 at 14 08 34" src="https://github.com/user-attachments/assets/cce5e348-bcbb-4cd6-a04e-e7dab7e378ee">
b4ff-ead77f82700e)

**Comment by talmo** [comment]
Hi @lustrauch,

Sorry for the delay, I think I was the one who was responsible for support when you posted your question 😅

> My first question is quite general: is it correct for a multiple animal models to use the same skeleton (same node points) for all the animals or should they be unique eg neck1 and neck2?

They should be the same skeleton.

> How do I get the tracking data from all the animals when they are not assigned IDs?

You should track them so the IDs get assigned -- this can be done by enabling the tracking option when you run inference. If you click over to the Instances tab, you'll see that they'll have tracks assigned. Check out [this part of the tutorial](https://sleap.ai/tutorials/proofreading.html).

The tracking probably won't be perfect though, so unless you don't mind mixing up the pups, you'll have to do some proofreading. Improving the poses will help with this to an extent, but looking at your screenshot, I think this will be pretty tough unfortunately.

> Do you have any suggestions to handle quite messy home cage tracking data? We are video monitoring the animals from P0-P21 to investigate how early life stress (using the limited bedding model) influences the behaviour of the pups and dam.

Yeah that's a toughie. We're doing a lot of home cage recording now too so I deeply appreciate the difficulty of tuning those setups for continuous tracking.

Our solution is pretty elaborate and not what I'd recommend for most people unless they plan to do a LOT of recording, but happy to find a time to chat if you shoot me an email --> talmo@salk.edu.

The TLDR is that we're using a combination of different visual markers to keep track of identities and a custom tracker (that will at some point make its way into SLEAP).


Cheers,

Talmo
